src/gradualizer_bin.erl

```diff
@@ -118,5 +118,5 @@ gcd1(A, 0) -> A;
 gcd1(A, B) ->
   case A rem B of
     0 -> B;
-    X -> gcd1(B, X)
+    X when X > 0 -> gcd1(B, X)
```

Comment:
The guard looks unnecessary. I know why it's there of course, but it's not ideal that we impose this coding style for users who want to avoid type errors.

Maybe we could override the type spec of `rem` in the prelude so it is closed under `non_neg_integer()`, e.g.

```Erlang
-spec 'rem'(non_neg_integer(), non_neg_integer()) -> non_neg_integer();
           (integer(), integer()) -> integer().
```

WDYT? Actually, I'm not even sure that we use specs for binary operators. Maybe it's handled in some other way.

Reply:
This seems quite useful and flexible, but I looked for it a bit and it seems that for operators, the "specs" are encoded directly in code as [`arith_op_arg_types`](https://github.com/josefs/Gradualizer/blob/bd92da1e47a0fdb8bd22009da10bd73f0422e066/src/typechecker.erl#L2609).

Reply:
That's true. Forget about this for now then.